src/main/scala/Column.scala

```diff
@@ -288,6 +366,708 @@ case class Column private (
+      s"${xmlSchema}date",
+      "invalid_date",
+      value
+    )
+  }
+
+  def processDateTimeDatatype(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}dateTime",
+      "invalid_datetime",
+      value
+    )
+  }
+
+  def processDateTimeStamp(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}dateTimeStamp",
+      "invalid_dateTimeStamp",
+      value
+    )
+  }
+
+  def processGDay(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gDay",
+      "invalid_gDay",
+      value
+    )
+  }
+
+  def processGMonth(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gMonth",
+      "invalid_gMonth",
+      value
+    )
+  }
+
+  def processGMonthDay(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gMonthDay",
+      "invalid_gMonthDat",
+      value
+    )
+  }
+
+  def processGYear(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gYear",
+      "invalid_gYear",
+      value
+    )
+  }
+  def processGYearMonth(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gYearMonth",
+      "invalid_gYearMonth",
+      value
+    )
+  }
+
+  def processTime(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}time",
+      "invalid_time",
+      value
+    )
+  }
+
+  def dateTimeParser(
+      datatype: String,
+      warning: String,
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    val dateFormatObject = DateFormat(format.flatMap(f => f.pattern), datatype)
+    dateFormatObject.parse(value) match {
+      case Right(value) => Right(value)
+      case Left(error)  => Left(ErrorWithoutContext(warning, error))
+    }
+  }
+
+  def addErrorIfRequiredValueAndValueEmpty(
+      value: String,
+      rowNumber: Long
+  ): Unit = {
+    if (required && value.isEmpty) {
+      errors :+= ErrorWithCsvContext(
+        "Required",
+        "schema",
+        rowNumber.toString,
+        columnOrdinal.toString,
+        value,
+        s"required => $required"
+      )
+    }
+  }
+
+  def validate(
+      value: String,
+      rowNumber: Long
+  ): WarningsAndErrors = {
+    if (nullParam.contains(value)) {
+      addErrorIfRequiredValueAndValueEmpty(value, rowNumber)
+    } else {
+      val values = separator match {
+        case Some(separator) => value.split(separator)
+        case None            => Array[String](value)
+      }
+      val datatypeBaseField = datatype.path("base")
+      val datatypeOfColumn = if (datatypeBaseField.isMissingNode) {
+        datatype.get("@id").asText()
```

Comment:
Again, it feels weird to be doing JSON parsing in here, can we add a property to the column which holds the baseDataType? And then populate that in the fromJson method?

Reply:
Done